### DISCUSSION TITLE
Allow external autolinks in titles of pull requests

### DISCUSSION THREAD

**Discussion by wolfpeyton** [community/community#9369] [Pull Requests]
[External autolinks](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/configuring-autolinks-to-reference-external-resources) can be active in pull request descriptions and comments, but not titles. It would be very nice if links were automatically substituted in titles as well. 

Thank you for your consideration.

**Comment by dkaiser-cfs** [comment]
+1

**Reply by mbonaci** [reply]
FYI for everyone who stumbles upon this, since no one from GitHub seems to be triaging these discussions, not even when they have 500 upvotes from the community:  

Install a Chrome extension called Alright [AutoLink References In GitHub Titles](https://chromewebstore.google.com/detail/blnmknhnaeeeaefecijcpaalckmcleam). 

No affiliation, here's what it does for me:

<img width="169" height="134" alt="image" src="https://github.com/user-attachments/assets/1ee180ee-f240-47c0-b8a0-dfff4c4baf1d" />

It also works when you open a PR - the configured substring of the title will be hyperlinked.

**Comment by tmarice** [comment]
+1

Though, it seems Github doesn't autolink anything in the PR title, including other PRs or issues.

**Comment by ivy-tian** [comment]
+1 when will you support this feature? Gitlab already support it

**Comment by kcritzsalient** [comment]
+1 this seems an obvious product need

**Comment by DSDmark** [comment]
+1

**Comment by kimljj** [comment]
This would be a really nice feature. As far as I know, gitlab already has this feature.

**Reply by AbirRazzak** [reply]
Can confirm, GitLab has this feature already

**Reply by alexander-williamson** [reply]
When you're reviewing a PR you'd be able to skip straight to your JIRA ticket to get more context (or other tool) via hyperlink. It's super useful in Bitbucket and is available in GitHub but you have to navigate to commits and it's a janky experience at the moment. We are asking for it to be in more places (like the PR title) so our quality of life as contributors/developers/reviewers improves

**Comment by ed-grow** [comment]
+1

**Comment by ryan-welsh** [comment]
+1

**Comment by martinslovacek** [comment]
Please consider implementing this feature in a foreseeable future. I believe it has a great outcome to effort ratio.

**Comment by nashashibi** [comment]
+1

**Comment by bdemmanou** [comment]
+1

**Comment by hashken** [comment]
+1

**Comment by jbrilhante-yogi** [comment]
+1
In my team, we prefix all our pull requests with the corresponding Jira issue. 
It would be really nice to have direct access to the Jira issue from the PR title without looking for the description, the commit messages or copy-pasting the Jira issue key.
Cheers! :wink:

**Comment by menzenski** [comment]
We migrated two years ago from Bitbucket to GitHub. Bitbucket supported this out-of-the-box and it's a feature I still greatly miss.

**Comment by mbonaci** [comment]
I have a feeling this would take 10 minutes to implement, but I guess no one at GH thinks it's important enough.

**Comment by tayyeb-ali** [comment]
+1 PLEASE implement this...our team prefixes ALL PR titles + ALL commit titles corresponding to the JIRA ticket.

**Comment by apetrosyan1613** [comment]
+1

**Comment by estefrare** [comment]
+1

**Comment by lupoalejandro** [comment]
+1

**Comment by q-ayush-gupta** [comment]
+1

**Comment by raghav99** [comment]
+1

**Comment by whoisjays** [comment]
+1

**Comment by jhearn-flutter-int** [comment]
+1

**Comment by leonteq-platc** [comment]
+1

**Comment by jthiesen** [comment]
Please implement this. We use prefix in PR titles for CRM links, and having an automatic link to that system is almost a must.

**Comment by dbaxiell** [comment]
+1

**Comment by akuhlman-cec** [comment]
+1

**Comment by GoglichidzeMike** [comment]
+1

**Comment by mbaranbe** [comment]
+1

**Comment by simplypixi** [comment]
+1

**Comment by paulweimer** [comment]
+1 Definitely needed. Right now we have to use "Autolink references" and always paste the Jira ticket (ABC-123) into the pull request description.

**Comment by dgpokl** [comment]
+1. Please.

**Comment by vubogovich** [comment]
+1, absolutely needed

**Comment by dave-amin** [comment]
+1, absolutely needed

**Comment by simon-slater-texada** [comment]
Please add this

**Comment by NesrenYousef** [comment]
Please add this feature

**Comment by mhschoen** [comment]
Jira integration would be great

**Comment by zendesk-shehata** [comment]
it is really important one .
+1

**Comment by Slauta** [comment]
I agree, +1

**Comment by suhas-chalotra** [comment]
+1

**Comment by liping-w** [comment]
+1
looking forward this good feature

**Comment by fukata** [comment]
+1

**Comment by MarcelMahdavian** [comment]
+1

**Comment by imasalygin** [comment]
+1

**Comment by alexgitcher** [comment]
+1

**Comment by hariths96** [comment]
+1

**Comment by vondermanzen** [comment]
+1

**Comment by pranavs1995** [comment]
+1
This will be really useful

**Comment by AlexandreCavanna** [comment]
+1

**Comment by Aymenworks** [comment]
+1
Please add!

**Comment by kentokento** [comment]
+1
🙏 🙏 🙏 🙏 🙏 🙏 🙏 🙏 🙏 🙏 🙏

**Comment by satorunooshie** [comment]
+1

**Comment by jbchr** [comment]
+1

**Comment by Charlie-Edmunds-SandboxVR** [comment]
+1

It's really much  more convenient to have it clickable from the title when prefixing the string. Especially if the title is pulled from a given submit to the branch automatically and the team is already prefixing it in the commit messages

**Comment by imelectronick** [comment]
+1

**Comment by fr-combo** [comment]
+1

**Comment by Jrgsto** [comment]
+1

**Comment by anyamiletic** [comment]
+1

**Comment by ruskinb1** [comment]
This Autolink is not working now, for ADO work items

**Comment by Zoey1988** [comment]
+1

**Comment by yst311** [comment]
+1

**Comment by kchenery** [comment]
How is this not implemented already?

**Comment by Vitalii4as** [comment]
+1
Had it in Bitbucket, after migration to GitHub, miss this feature a lot.

**Comment by chunfeilung** [comment]
Our team ran into the same issue after migrating to GitHub from Bitbucket. At first we also tried to work around it by manually pasting links to JIRA in our pull request descriptions, but eventually I ended up writing [a Chrome extension](https://chromewebstore.google.com/detail/alright-autolink-referenc/blnmknhnaeeeaefecijcpaalckmcleam) that – at least for now – does what GitHub should have done.

**Comment by skrollme** [comment]
+1

**Comment by marcos-aloware** [comment]
+1

**Comment by lsiebels** [comment]
+1

**Comment by FinnReinhardtBsc** [comment]
+2

**Comment by leijer** [comment]
+3

**Comment by mbonaci** [comment]
+4

**Comment by SeanLively** [comment]
This is the only "feature" we're missing from Bitbucket.

**Comment by alexander-williamson** [comment]
Still wanting this feature please, it is a clear requirement of this feature

**Comment by RustamSultansoy** [comment]
+1

**Comment by martaEntur** [comment]
+1

**Comment by c3-sxc** [comment]
+1

**Comment by robertvanduursen** [comment]
+1

**Comment by kokchuan101** [comment]
+1

**Comment by siaavush** [comment]
+1

**Comment by afedulov** [comment]
+1

**Comment by tyronejosee** [comment]
+1

**Comment by ChannyIntelcom** [comment]
+1

**Comment by alzix** [comment]
+1

**Comment by Nihileon** [comment]
+1

**Comment by johnisom** [comment]
+1

**Comment by tombiddulph** [comment]
+1

**Comment by simon-slater-texada** [comment]
Please add this feature

**Comment by TheFriendlyCoder** [comment]
+1

**Comment by afedulov** [comment]
While this issue keeps collecting dust, here is a workaround that solved it for me in Chrome:
https://chromewebstore.google.com/detail/alright-autolink-referenc/blnmknhnaeeeaefecijcpaalckmcleam

It offers exactly the same settings as standard GitHub autolinks, but applies them to PR titles:

<img width="641" alt="image" src="https://github.com/user-attachments/assets/a8498ad3-3587-496a-999b-72ea5697e302" />

**Reply by mbonaci** [reply]
Nice, but it probably requires giving the author access to our private repo.

**Reply by alechemy** [reply]
> Nice, but it probably requires giving the author access to our private repo.

It does not. It's just adding links to the page, after all. 
https://github.com/chunfeilung/alright

**Reply by mbonaci** [reply]
The extension works great!
BTW Even though the source is open, a malicious author could probably use an extension to gain access to your private repo (at least to the content of the pages being browsed), given that Chrome auto-updates extensions.  
Sounds a bit paranoid, I know, but...

**Reply by alechemy** [reply]
In my experience Chrome alerts me when I've updated an extension and it's requesting new permissions. https://developer.chrome.com/docs/extensions/develop/concepts/permission-warnings

But yes, point taken. If you were really worried, you could always clone/fork the repo and build the extension yourself?

**Reply by afedulov** [reply]
To avoid such surprises you can always fork the repo and install manually using "developer mode". I had to do it anyways to allow access to an enterprise instance https://github.com/chunfeilung/alright/blob/main/manifest.json#L16

**Comment by tfactor2** [comment]
I created a TamperMonkey script to address the issue:
```js
// ==UserScript==
// @name         github-pr-jira-link-injector
// @version      0.1
// @description  Adds a link from Github PRs to Jira issues; the link should follow the following convention: ABC-12345
// @author       nizametdinov
// @match        https://github.com/*/pull/*
// @icon         https://www.google.com/s2/favicons?sz=64&domain=github.com
// @grant        none
// ==/UserScript==

(function injectLink() {
    const $issueTitle = document.querySelector('.js-issue-title, [data-component="PH_Title"] .markdown-title');
    // always track changes as the contents of the page may be changed, say, by switching tabs (Conversation, Commits, Files changed, etc)
    if ($issueTitle && !$issueTitle.hasAttribute('data-github-pr-jira-link-injector-processed')) {
        const jiraKey = /[A-Z]+\-\d+/.exec($issueTitle.innerText);
        if (jiraKey) {
            $issueTitle.innerHTML = `<a href="https://<your-domain>.atlassian.net/browse/${jiraKey}">${$issueTitle.innerText}</a>`;
        }
        $issueTitle.setAttribute('data-github-pr-jira-link-injector-processed', 'true');
    }
    setTimeout(injectLink, 1000);
})();
```

**Comment by shimonenator** [comment]
This issue is missing an `Enhancement` label

**Reply by wolfpeyton** [reply]
As the creator of the discussion, is there a way for me to add this label? It seems like I don't have permissions to do so.

**Comment by ugocottin** [comment]
+1

**Comment by lipsumar** [comment]
+1

**Comment by ReneLohrer** [comment]
+1

**Comment by milov-dmitriy** [comment]
+1 its must have <3

**Comment by prasathsarath** [comment]
+1

**Comment by tarunmittal24** [comment]
+1

**Comment by PabloMartinezAngstrom** [comment]
+1

**Comment by tabuchid** [comment]
+1

**Comment by poseggaQM** [comment]
+1

**Comment by ph-gh-andrew-kozhokaru** [comment]
+1

**Comment by zabetak** [comment]
Definitely useful +1

**Comment by attluh** [comment]
+1 - Bitbucket and Gitlab both have this feature. My team prefixes all of our PRs with Jira ticket numbers. It would be so helpful.

**Reply by mbonaci** [reply]
I wouldn't get my hopes too high. It doesn't seem like anyone from GH is watching this space. Zero feedback in the last 4 years.  
I'm using the Chrome extension mentioned here: https://github.com/orgs/community/discussions/9369#discussioncomment-11820433

**Comment by mbonaci** [comment]
FYI I've just posted this question about this feature request: https://github.com/orgs/community/discussions/161604

**Comment by alzix** [comment]
we ended up adding a job that appends the ticket number to the PR description body when the PR is created.
External links work for the PR body

**Comment by nosu** [comment]
+1

**Comment by horavamartin** [comment]
+1

**Comment by fviken** [comment]
+1

**Comment by szreniak** [comment]
+1

**Comment by ocvagabond-tech** [comment]
+1

**Comment by AnubhavSethiDynamite** [comment]
+1
Still dont understand why this is not implemented since last 4 years. This is a must have feature

**Comment by MichaelChirico** [comment]
+1 to what seems like an obvious, low-hanging fruit

I feel like a scold chasing outside contributors to "please add references to the description, not the title" because of this missing feature. It seems obvious GitHub should "just work" where authors assume it will.

**Reply by mbonaci** [reply]
Install Alright: [Autolink references in GitHub titles](https://chromewebstore.google.com/detail/blnmknhnaeeeaefecijcpaalckmcleam).  
You'll thank me later ;)  
No one from Microsoft seems to be watching these discussions these days.

<img width="169" height="134" alt="image" src="https://github.com/user-attachments/assets/87a24737-cb44-46e4-aec2-d1e0dd73a9d5" />

**Comment by twojnarowski** [comment]
I agree, it should work in titles of PRs as well, both on the list of PR and in the details of PR

**Comment by jmanuelmartinez** [comment]
+1

Missing this feature that is by default available in ADO... what's happening Microsoft? Don't you like your new toy? 🥲

**Comment by mwessman-amd** [comment]
+1

**Reply by mwessman-amd** [reply]
It seems like such a simple addition too. The `autolink references` feature already exists, they just need to enable it for the PR titles. Any Microsoft intern could fix this in a few hours.

**Comment by dhiogoboza-ra** [comment]
+1

**Comment by SoLasseVanttinen** [comment]
+1

**Comment by Nigiffr** [comment]
+1

**Comment by hnwh224** [comment]
+1

**Comment by dhiogoboza-ra** [comment]
+1

**Comment by mariokresic** [comment]
+1

**Comment by EV-LFJ** [comment]
+1

**Comment by kaspermoerch** [comment]
+1

**Comment by dylanredmond-fbg** [comment]
Any progress on this? [GitLab has this functionality](https://docs.gitlab.com/integration/jira/issues/)

**Comment by lukamrko** [comment]
+1

**Comment by william111ttt** [comment]
I agree—having auto-links work in titles as well would make things more consistent and easier to read. Thanks for the suggestion!

**Comment by andreakostakis-veolia** [comment]
+1

**Comment by fholznagel** [comment]
+1

**Comment by k-risc** [comment]
+1
